.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,47 @@
+name: On Pull Request
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+    env:
+      TEST_AUTH_TOKEN: ${{ secrets.ALPHA_TEST_AUTH_TOKEN }}
+      TEST_CACHE_NAME: python-integration-test-cache
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Bootstrap poetry
+        run: |
+          curl -sL https://install.python-poetry.org | python - -y
```

Comment:
@malandis dont need to block this PR but I do generally like to pin versions where ever possible. Also kinda hate piping curl to bash. Any other easy alternate ways to install and pin to specific version?

Reply:
Agreed about pinning versions, but... This is the recommended installation method, _and_ the method the poetry maintainers use in their own Github actions, so I'm not concerned.